src/biogeochem/VOCEmissionMod.F90

```diff
@@ -818,90 +811,52 @@ function get_gamma_L(fsun240_in,elai_in)
-    use clm_varpar , only : nlevsoi
-    !
-    ! !ARGUMENTS:
-    implicit none
-    real(r8),intent(in) :: clayfrac_in
-    real(r8),intent(in) :: sandfrac_in
-    real(r8),intent(in) :: h2osoi_vol_in(nlevsoi)
-    real(r8),intent(in) :: h2osoi_ice_in(nlevsoi)
-    real(r8),intent(in) :: dz_in(nlevsoi)
-    real(r8),intent(in) :: bsw_in(nlevsoi)
-    real(r8),intent(in) :: watsat_in(nlevsoi)
-    real(r8),intent(in) :: sucsat_in(nlevsoi)
-    real(r8),intent(in) :: root_depth_in
-    !
-    ! !LOCAL VARIABLES:
-    real(r8)            :: get_gamma_SM
-    integer  :: j
-    real(r8) :: nl                      ! temporary number of soil levels
-    real(r8) :: theta_ice               ! water content in ice in m3/m3
-    real(r8) :: wilt                    ! wilting point in m3/m3
-    real(r8) :: theta1                  ! temporary
-    real(r8), parameter :: deltheta1=0.06_r8               ! empirical coefficient
-    real(r8), parameter :: smpmax = 2.57e5_r8              ! maximum soil matrix potential
-    !-----------------------------------------------------------------------
+  function get_gamma_SM(btran_in)
 
-    if ((clayfrac_in > 0) .and. (sandfrac_in > 0)) then 
-       get_gamma_SM = 0._r8
-       nl=0._r8
-       
-       do j = 1,nlevsoi
-          if  (sum(dz_in(1:j)) < root_depth_in)  then
-             theta_ice = h2osoi_ice_in(j)/(dz_in(j)*denice)
-             wilt = ((smpmax/sucsat_in(j))**(-1._r8/bsw_in(j))) * (watsat_in(j) - theta_ice)
-             theta1 = wilt + deltheta1
-             if (h2osoi_vol_in(j) >= theta1) then 
-                get_gamma_SM = get_gamma_SM + 1._r8
-             else if ( (h2osoi_vol_in(j) > wilt) .and. (h2osoi_vol_in(j) < theta1) ) then
-                get_gamma_SM = get_gamma_SM + ( h2osoi_vol_in(j) - wilt ) / deltheta1
-             else
-                get_gamma_SM = get_gamma_SM + 0._r8
-             end if
-             nl=nl+1._r8
-          end if
-       end do
-       
-       if (nl > 0._r8) then
-          get_gamma_SM = get_gamma_SM/nl
-       endif
+    !---------------------------------------
+    ! May 22, 2024
+    ! Activity factor for soil moisture of Isoprene (Wang et al., 2022, JAMES)
+    ! It is based on eq. (11) in the paper. Because the temperature response
+    ! of isoprene has been explicitly included in CLM;
 
-       if (get_gamma_SM > 1.0_r8) then 
-          write(iulog,*) 'healdSM > 1: gamma_SM, nl', get_gamma_SM, nl
-          get_gamma_SM=1.0_r8
-       endif
+    !ARGUMENTS:
+    implicit none
+    real(r8),intent(in) :: btran_in
+
+    !!!------- the drought algorithm--------
+    real(r8), parameter :: a1 = -7.4463_r8
+    real(r8), parameter :: b1 = 3.2552_r8
+    real(r8)            :: get_gamma_SM
+    !---------------------------------------
 
+    if (btran_in >= 1._r8) then
+       get_gamma_SM = 1._r8
     else
-       get_gamma_SM = 1.0_r8
-    end if
+       get_gamma_SM = 1._r8 / (1._r8 + b1 * exp(a1 * (btran_in - 0.2_r8)))
```

Comment:
In this formula, I could see naming the 0.2 to something like btran_threshold or whatever you think is appropriate.

Reply:
Hi @slevis-lmwg, btran_threshold is a good definition for this! Thank you.